File: android/app/src/main/java/com/liive/ride/designsystem/LiiveBadge.kt
//  LiiveBadge.kt  ·  Liive Ride DS (Compose)  ·  mirrors components/core/Badge
package com.liive.ride.designsystem

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp

enum class BadgeColor { Neutral, Accent, Success, Warning, Danger, Info }

@Composable
fun LiiveBadge(text: String, color: BadgeColor = BadgeColor.Neutral, solid: Boolean = false, dot: Boolean = false) {
    val c = LiiveTheme.colors
    val fg = when (color) {
        BadgeColor.Neutral -> c.textSecondary; BadgeColor.Accent -> c.accent
        BadgeColor.Success -> c.success; BadgeColor.Warning -> c.warning
        BadgeColor.Danger -> c.danger; BadgeColor.Info -> c.info
    }
    val solidBg = when (color) {
        BadgeColor.Neutral -> c.fill; BadgeColor.Accent -> c.accent
        BadgeColor.Success -> c.success; BadgeColor.Warning -> c.warning
        BadgeColor.Danger -> c.danger; BadgeColor.Info -> c.info
    }
    val tintBg = when (color) {
        BadgeColor.Neutral -> c.fillTertiary; BadgeColor.Accent -> c.accentTint
        BadgeColor.Success -> c.successTint; BadgeColor.Warning -> c.warningTint
        BadgeColor.Danger -> c.dangerTint; BadgeColor.Info -> c.infoTint
    }
    val onSolid = if (color == BadgeColor.Warning || color == BadgeColor.Info) Color.Black else Color.White
    val content = if (solid) onSolid else fg

    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(LiiveBadgeLayout.ContentGap),
        modifier = Modifier
            .clip(CircleShape)
            .background(if (solid) solidBg else tintBg)
            .padding(horizontal = LiiveBadgeLayout.HorizontalPadding, vertical = LiiveBadgeLayout.VerticalPadding)
    ) {
        if (dot) {
            Box(
                Modifier
                    .size(LiiveBadgeLayout.DotSize)
                    .clip(CircleShape)
                    .background(content)
            )
        }
        Text(
            text,
            color = content,
            style = MaterialTheme.typography.labelMedium.copy(
                fontWeight = FontWeight.SemiBold,
                letterSpacing = LiiveBadgeLayout.LetterSpacing,
                lineHeight = LiiveBadgeLayout.LineHeight
            )
        )
    }
}

private object LiiveBadgeLayout {
    val ContentGap = LiiveSpacing.xs + LiiveSpacing.xs2 / 2
    val DotSize = LiiveSpacing.s - LiiveSpacing.xs2 / 2
    val HorizontalPadding = LiiveSpacing.s + LiiveSpacing.xs2 / 2
    val VerticalPadding = LiiveSpacing.xs - LiiveSpacing.xs2 / 2
    val LetterSpacing = (LiiveSpacing.xs2.value / 20).sp
    val LineHeight = (LiiveSpacing.m.value * LineHeightMultiplier).sp
    private const val LineHeightMultiplier = 1.3f
}

File: Packages/RideSharingFeature/Sources/RideSharingFeature/DesignSystem/LiiveBadge.swift
//  LiiveBadge.swift  ·  Liive Ride DS (SwiftUI)
//  Capsule status/metadata badge. Mirrors components/core/Badge.

import SwiftUI

public struct LiiveBadge: View {
    public enum Color { case neutral, accent, success, warning, danger, info }
    let text: String
    var color: Color = .neutral
    var solid: Bool = false
    var dot: Bool = false

    public init(_ text: String, color: Color = .neutral, solid: Bool = false, dot: Bool = false) {
        self.text = text; self.color = color; self.solid = solid; self.dot = dot
    }

    private var fg: SwiftUI.Color {
        switch color {
        case .neutral: return LiiveColor.textSecondary
        case .accent: return LiiveColor.accent
        case .success: return LiiveColor.success
        case .warning: return LiiveColor.warning
        case .danger: return LiiveColor.danger
        case .info: return LiiveColor.info
        }
    }
    private var solidBg: SwiftUI.Color {
        switch color {
        case .neutral: return LiiveColor.fill
        case .accent: return LiiveColor.accent
        case .success: return LiiveColor.success
        case .warning: return LiiveColor.warning
        case .danger: return LiiveColor.danger
        case .info: return LiiveColor.info
        }
    }
    private var tintBg: SwiftUI.Color {
        switch color {
        case .neutral: return LiiveColor.fillTertiary
        case .accent: return LiiveColor.accentTint
        case .success: return LiiveColor.successTint
        case .warning: return LiiveColor.warningTint
        case .danger: return LiiveColor.dangerTint
        case .info: return LiiveColor.infoTint
        }
    }
    private var onSolid: SwiftUI.Color {
        (color == .warning || color == .info) ? .black : .white
    }

    public var body: some View {
        HStack(spacing: LiiveBadgeLayout.contentGap) {
            if dot {
                Circle()
                    .fill(solid ? onSolid : fg)
                    .frame(width: LiiveBadgeLayout.dotSize, height: LiiveBadgeLayout.dotSize)
            }
            Text(text)
                .font(LiiveFont.caption1.weight(.semibold))
                .tracking(LiiveBadgeLayout.letterSpacing)
        }
        .padding(.horizontal, LiiveBadgeLayout.horizontalPadding)
        .padding(.vertical, LiiveBadgeLayout.verticalPadding)
        .foregroundColor(solid ? onSolid : fg)
        .background(solid ? solidBg : tintBg)
        .clipShape(Capsule())
    }
}

private enum LiiveBadgeLayout {
    static let contentGap = LiiveSpacing.xs + LiiveSpacing.xs2 / 2
    static let dotSize = LiiveSpacing.s - LiiveSpacing.xs2 / 2
    static let horizontalPadding = LiiveSpacing.s + LiiveSpacing.xs2 / 2
    static let verticalPadding = LiiveSpacing.xs - LiiveSpacing.xs2 / 2
    static let letterSpacing = LiiveSpacing.xs2 / 20
}
